src/lib/y2network/connection_config/base.rb

```diff
@@ -90,6 +90,7 @@ def initialize
       # @return [Boolean] true when both connections are same
       #                   false otherwise
       def ==(other)
+        return false if other.nil?
```

Comment:
what about more generic `return false if self.class != other.class`? this way it also handle comparison of different types.